trunk/SUAVE/Methods/Propulsion/scramjet_sizing.py

```diff
@@ -0,0 +1,158 @@
+## @ingroup Methods-Propulsion
+# scramjet_sizing.py
+# 
+# Created:  April 2018, W. Maier
+# Modified:        
+#           
+
+# ----------------------------------------------------------------------
+#   Imports
+# ----------------------------------------------------------------------
+import SUAVE
+import numpy as np
+from SUAVE.Core import Data
+
+# ----------------------------------------------------------------------
+#   Sizing
+# ----------------------------------------------------------------------
+## @ingroup Methods-Propulsion
+def scramjet_sizing(scramjet,mach_number = None, altitude = None, delta_isa = 0, conditions = None):  
+    """ This function sizes a scramjet for the input design conditions.
+    """    
+    
+    #Unpack components
+    #check if altitude is passed or conditions is passed
+    if(conditions):
+        #use conditions
+        pass
+        
+    else:
+        #check if mach number and temperature are passed
+        if(mach_number==None or altitude==None):
+            
+            #raise an error
+            raise NameError('The sizing conditions require an altitude and a Mach number')
+        
+        else:
+            #call the atmospheric model to get the conditions at the specified altitude
+            atmosphere = SUAVE.Analyses.Atmospheric.US_Standard_1976()
+            atmo_data = atmosphere.compute_values(altitude,delta_isa,True)          
+            
+            p   = atmo_data.pressure          
+            T   = atmo_data.temperature       
+            rho = atmo_data.density          
+            a   = atmo_data.speed_of_sound    
+            mu  = atmo_data.dynamic_viscosity   
+        
+            # setup conditions
+            conditions = SUAVE.Analyses.Mission.Segments.Conditions.Aerodynamics()            
+        
+            # freestream conditions    
+            conditions.freestream.altitude                    = np.atleast_1d(altitude)
+            conditions.freestream.mach_number                 = np.atleast_1d(mach_number)
+            conditions.freestream.pressure                    = np.atleast_1d(p)
+            conditions.freestream.temperature                 = np.atleast_1d(T)
+            conditions.freestream.density                     = np.atleast_1d(rho)
+            conditions.freestream.dynamic_viscosity           = np.atleast_1d(mu)
+            conditions.freestream.gravity                     = np.atleast_1d(9.81)
+            conditions.freestream.isentropic_expansion_factor = np.atleast_1d(scramjet.working_fluid.compute_gamma(T,p))
```

Comment:
I know this is a common oversight in other parts of the code, but please provide a not hard coded value for gravity.

Reply:
Still needed

Reply:
Done